### DISCUSSION TITLE
select 下拉菜单，有时不显示下拉菜单，导致不能搜索

### DISCUSSION THREAD

**Discussion by wangwenjie7** [layui/layui#1847] [问答]
例如，输入框已有文字，全选输入框的内容，直接打字，就不能进行搜索，也没弹出下拉菜单，重新点一下输入框，才能激活下拉菜单，然后才能搜索。这个有方法可以避免吗？我看了官方给的select例子中，也是这种情况。
![20240430-112840](https://github.com/layui/layui/assets/29366022/93573842-6f9b-4867-a0c6-9987d3eb4f2c)

**Comment by Sight-wcg** [comment]
有详细的重现步骤吗？官网快速试了下，没发现这个问题。

**Reply by wangwenjie7** [reply]
我这个截图，就是官网form例子里面的；步骤就是：先选一个，然后鼠标从后往前，把选中的文字全选，直接打字，这个时候，就不弹出下拉菜单了，也无搜索功能。（我上面标题可能说的有些问题，不删除，选中直接打字替换）

**Reply by hjingsuper** [reply]
![2024-04-30-13-54-40](https://github.com/layui/layui/assets/36070786/601b5ef9-3c46-44a1-ae00-23604ddff641)
按照你说的步骤，是这样么？没发现你说的问题额

**Reply by wangwenjie7** [reply]
上面的步骤基本是对的，就是鼠标从后往前选中文字的时候，鼠标松开的位置，在框框外面

**Reply by wangwenjie7** [reply]
@Sight-wcg @hjingsuper 这样复现：
![select选项](https://github.com/layui/layui/assets/29366022/2736a902-43f9-4ae1-8e20-4512790d16f5)

**Comment by wangwenjie7** [comment]
如下所示
![select选项](https://github.com/layui/layui/assets/29366022/6c0efaba-e115-4886-a260-479d025d4a82)

**Reply by LSL1618** [reply]
@wangwenjie7 ，还真是，鼠标选中搜索内容，然后离开输入框，接着直接打字，搜索功能就失效了，下拉也不弹出了。这个bug好奇怪，输入框焦点明明还在，只是鼠标指针不在输入框区域内，总不能搜索功能和下拉面板与鼠标指针的位置有关系吧。

**Reply by bxjt123** [reply]
测试了下即使不先选中某项（输入框为空未获得焦点时），在单击输入框时不松开鼠标移出到框外松开同样有焦点但不弹下拉框。
看了下源码判断的是`click`时显示下拉框，然而鼠标移出`input`时才松开是不触发`click`事件，如果选中一定会触发`select`事件，也许判断可以加个`select`或用`mousedown/focus`估计更好点，但不好说是否会引入其他bug
